File: src/main/scala/org/joshjoyce/lumivore/sync/UploadLogImporter.scala
package org.joshjoyce.lumivore.sync

import org.joshjoyce.lumivore.db.SqliteDatabase
import scala.io.Source
import java.sql.SQLException

object UploadLogImporter {
  val key = "INFO  GlacierUploader - "
  val keyLen = key.size

  def main(args: Array[String]) {
    if (args.isEmpty) {
      println("No file given")
      System.exit(-1)
    }
    val filename = args(0)
    val database = new SqliteDatabase
    database.connect()

    val source = Source.fromFile(filename)

    source.getLines().foreach {
      line => {
        if (line.contains(key)) {
          val pipeIndex = line.indexOf('|', keyLen)
          val path = line.substring(keyLen, pipeIndex)
          val archiveId = line.substring(pipeIndex + 1)
          val syncs = database.getSync(path)
          println(path)
          syncs.headOption.foreach {
            case (_, sha1, _) => try {
              database.insertUpload(sha1, "photos", archiveId)
            } catch {
              case (e:SQLException) if e.getMessage.contains("CONSTRAINT") => {}
            }
          }
        }
      }
    }
  }
}

File: src/main/java/org/joshjoyce/lumivore/sync/UploadLogImporter.java
package org.joshjoyce.lumivore.sync;

import org.joshjoyce.lumivore.db.SqliteDatabase;

import java.io.File;
import java.nio.file.Files;

public class UploadLogImporter {
    private static final String key = "INFO  GlacierUploader - ";
    private static final int keyLen = key.length();

    public static void main(String[] args) throws Exception {
        if (args.length == 0) {
            System.out.println("No file given");
            System.exit(-1);
        }
        var filename = args[0];
        var database = new SqliteDatabase();
        database.connect();

        var lines = Files.readAllLines(new File(filename).toPath());
        lines.forEach(line -> {
            if (line.contains(key) && line.contains("/media")) {
                var pipeIndex = line.indexOf('|', keyLen);
                var pathStart = line.indexOf("/");
                if (pathStart >= 0)
                {
                    var path = line.substring(pathStart, pipeIndex);
                    var archiveId = line.substring(pipeIndex + 1);
                    var syncs = database.getSync(path);
                    //System.out.println(path);

                    if (syncs != null)
                    {
                        try
                        {
                            database.insertUpload(syncs.hash, "photos", archiveId);
                        } catch (Exception e)
                        {
                            if (!e.getMessage().contains("CONSTRAINT"))
                            {
                                e.printStackTrace();
                            }
                        }
                    } else
                    {
                        System.out.println("NOSYNC|" + path);
                    }
                }
            }
        });
    }
}

